### PR TITLE
New thumbnails for General-purpose images and scientific images

### PR DESCRIPTION
**Pull request by kolibril13**
## Description

Fixes https://github.com/scikit-image/scikit-image/issues/4384#
This PR adds a nice thumbnail for the General-purpose images:
image

I used the variable `sphinx_gallery_thumbnail_number = -1` (see https://sphinx-gallery.github.io/stable/auto_examples/plot_4_choose_thumbnail.html) , so  sphinx will recognize the right image for the thumblail. 
I am converting it from draft to real PR as soon as the docs build looks good.
@grlee77: Could you review this pr?
## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: doc/examples/data/plot_general.py [hunk 1]
```diff
@@ -45,3 +45,28 @@
...
 plt.show()
+
+
+"""
+Thumbnail image for the gallery
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+from skimage import data
```
Comment (grlee77) [inline doc/examples/data/plot_general.py:56]: Can you move the numpy import after the matplotlib ones above (the other two here are duplicate imports)

```suggestion
```

File: doc/examples/data/plot_general.py [hunk 2]
```diff
@@ -45,3 +45,28 @@
...
+# The next line sets the thumbnail for the last figure in the gallery
+# sphinx_gallery_thumbnail_number = -1
```
Comment (grlee77) [inline doc/examples/data/plot_general.py:58]:
```suggestion
# sphinx_gallery_thumbnail_number = -1
```

**Review by grlee77 — CHANGES_REQUESTED** [review]
The suggestions here are the ones the pep8 bot was warning about. Otherwise this looks good (pending final rendered doc check)

**Comment by grlee77** [comment]
Not sure if there is good way to not also have the thumbnail appear in the gallery itself? It looks like the other option is to [store a static image and then point to that for the thumbnail](https://sphinx-gallery.github.io/stable/auto_examples/plot_4b_provide_thumbnail.html#providing-a-figure-for-the-thumbnail-image). That is problematic if our datasets change, though (rare, but has happened a couple of times in the past).

**Comment by kolibril13** [comment]
I don't like too much to add static images to git as it will increase the repo size and I think it's nice to have this auto-updating option. Manually re-creating the thumbnail everytime the data changes feels like extra work, which would also have to be kept in mind. By auto-generating the thumbnail, less manual work would be required, and a good side effect is that there is a new example of how to create a grid with three dots that indicate further images.
The doc artifact preview looks nice:
image
However, the example looks a bit odd, as it is one big example:
https://3680-2014929-gh.circle-artifacts.com/0/doc/build/html/auto_examples/data/plot_general.html#sphx-glr-auto-examples-data-plot-general-py
I ll try to split it into two examples, similar to here: https://scikit-image.org/docs/stable/auto_examples/registration/plot_stitching.html?highlight=stitching
I guess, I just have to add this syntax:
```py
############################################################################
# Thumbnail image for the gallery
```

**Comment by kolibril13** [comment]
Do I have to do something to trigger the Github Actions? It seems they were not triggered by the last commit
image

**Comment by kolibril13** [comment]
Here is the latest version: https://3683-2014929-gh.circle-artifacts.com/0/doc/build/html/auto_examples/index.html
I think this is now ready for review.
Two more advantages of keeping the code for the thumbail examples in the docs:

* It becomes clear that images can be accessed in this object oriented way
* The functions are clickable and directly linked to the place where more information about the images is provided
image

**Comment by grlee77** [comment]
> https://3683-2014929-gh.circle-artifacts.com/0/doc/build/html/auto_examples/index.html

Yeah, this looks better. There is much clearer separation from the individual example images now.

I do think the 3 x 3 grid looks pretty nice as well. I also liked your suggestion to do the same thing for the scientific images as well. It may be easiest to just also add that here, but also fine if you want to do it in a separate PR.

**Review by grlee77 — APPROVED** [review]
I have approved the PR, but will leave open for a bit to see if there is any other feedback. If you want to play with switching to 3x3 or updating the scientific images as well, please go ahead

**Comment by kolibril13** [comment]
I've changed to the 3x3 example now.
And I will do the scientific images images thumbnail in a seperate pr after this one is merged.

**Comment by kolibril13** [comment]
Seems like Github has a lot of problems today, error this time: "Artifact not found"  I will run the CI again.

**Review by jni — APPROVED** [review]

**Comment by kolibril13** [comment]
Latest version:
https://output.circle-artifacts.com/output/job/9bb49f1a-8b71-4f48-832f-884648e5d3f2/artifacts/0/doc/build/html/auto_examples/index.html
image
I will increase the spacing for the scientific images a bit

**Comment by kolibril13** [comment]
https://output.circle-artifacts.com/output/job/463790b2-ec5e-4f72-a0af-53327f6f4e53/artifacts/0/doc/build/html/auto_examples/index.html
I think now it looks great!
image
But the specific images could also be tweaked, what about this new thumbnail?
image
(last one is generated with)
```py

from matplotlib.offsetbox import AnchoredText

ax1 = plt.subplot(2,2,1)
ax1.imshow(left_motorcycle)
ax1.add_artist(AnchoredText("Stereo", prop=dict(size=20), frameon=True , borderpad=0, loc='upper left'))
ax2 = plt.subplot(2,2,2)
ax2.imshow(left_vortex)
ax2.add_artist(AnchoredText("PIV", prop=dict(size=20), frameon=True , borderpad=0, loc='upper left'))

ax3 = plt.subplot(2,4,5)
ax3.imshow(im1, cmap="gray")
ax4 = plt.subplot(2,4,6)
ax4.imshow(im2, cmap="gray")
ax5 = plt.subplot(2,4,7)
ax5.imshow(im3, cmap="gray")
ax6 = plt.subplot(2,4,8)
ax6.imshow(im4, cmap="gray")
for ax in [ax1,ax2,ax3,ax4,ax5,ax6]:
    ax.axis("off")
plt.tight_layout()
```

**Review by grlee77 — APPROVED** [review]
LGTM!

File: doc/examples/data/plot_general.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 """
 import matplotlib.pyplot as plt
 import matplotlib
-
+import numpy as np
```
Comment (lagru) [inline doc/examples/data/plot_general.py:11]:
```suggestion
import numpy as np

```

File: doc/examples/data/plot_scientific.py [hunk 4]
```diff
@@ -37,3 +38,28 @@
         plt.imshow(image)
 
 plt.show()
+
```
Comment (lagru) [inline doc/examples/data/plot_scientific.py:41]:
```suggestion
```

File: doc/examples/data/plot_general.py [hunk 5]
```diff
@@ -45,3 +45,25 @@
...
+axs[1, 0].imshow(data.colorwheel())
+axs[1, 1].imshow(data.camera(), cmap=plt.cm.gray)
+axs[1, 2].imshow(data.cat())
+axs[2, 0].imshow(data.checkerboard(), cmap=plt.cm.gray)
+axs[2, 1].imshow(data.clock(), cmap=plt.cm.gray)
+further_img = np.full((300, 300), 255)
+for xpos in [100, 150, 200]:
+    further_img[150 - 10 : 150 + 10, xpos - 10 : xpos + 10] = 0
+axs[2, 2].imshow(further_img, cmap=plt.cm.gray)
+plt.subplots_adjust(wspace=0.1, hspace=0.1)
```
Comment (lagru) [inline doc/examples/data/plot_general.py:69]:
```suggestion
plt.subplots_adjust(wspace=0.1, hspace=0.1)

```

**Review by lagru — COMMENTED** [review]
Added some style changes / suggestions so that the PEP8 bot is satisfied.

**Comment by lagru** [comment]
Just saw that you already opened #6301 :smile:

**Pull request merged**